### ISSUE TITLE
fenced code block not syntax highlighted in marked down

### ISSUE THREAD

**Issue by laoshaw** [sotte/presenting.vim#37]
I have ` ```cpp ``` ` for c++, and c for c, and python for python etc, i.e. the 'standard' markdown code block, however they're either displayed as a single color, or the code just disappears, the code block in markdown does not work for me on ubuntu 20.04 vim8.

**Comment by sotte** [comment]
@laoshaw thanks for reporting!

I was looking into other issues and can confirm that code block are not syntax highlighted.

Please provide a md snippet which contains an example in which the code is not displayed at all.

**Comment by sotte** [comment]
Note so self:
maybe related to this: https://github.com/sotte/presenting.vim/commit/6d87e549fb0bd00db908f501c7b4bff7436daa05

**Comment by laoshaw** [comment]
A markdown snippet:

`
## hello world for c++

```cpp
#include <iostream>
int main(){ std::cout << "hello world" << std::endl;
```

## hello world for c

```c
#include <stdio.h>
int main(void) { printf("hello world\n");} 
```
`

**Comment by sotte** [comment]
Haha, the `#include` is interpreted as a heading and creates a new slide :)
The snippet you posted creates 4 slides, not 2 as one would expect.
Fixing the separator is easy, but I have to investigate the syntax hl issue

**Comment by sotte** [comment]
@laoshaw just to have a semi working version this would be the regex for the slid separator:

```vim
# ftplugin/markdown.vim
let b:presenting_slide_separator_default = '\v(^|\n)\ze#{1,2} '
```

You should be able to overwrite the separator like this:
https://github.com/sotte/presenting.vim#configuration

**Comment by laoshaw** [comment]
Thanks. That fixed the header issue. Hope syntax can be highlighted somehow.

A screenshot of one slide is attached, there are two bars not sure if it's be design. plus the two lines for code block, e.g.

    ```c
    ```

ideally should not be visible.

![x](https://user-images.githubusercontent.com/18351761/120497454-09a77180-c384-11eb-9e83-414d5d549271.png)

**Comment by sotte** [comment]
The bars are there on purpose, fencing off the code

> ideally should not be visible.

That is/should be the case if it works.

**Comment by laoshaw** [comment]
If code can be highlighted they will stand out visibly enough without the two bars I feel. It would be nice to make the two bars optional as they're a bit heavy visually when I have say two or three small snippets of code on one slide(that will be 6 bars then)

**Comment by sotte** [comment]
Yeah, design is subjective :) Initially presenting did not do any formatting, it just used the existing hl.

If you want a super lightweight version of presenting you can use an old version which does not use figlet, but does proper syntax hl:
```
git checkout 9fb8ccc
```

Then "fix" the slide sep:
```
au FileType markdown let s:presenting_slide_separator = '\v(^|\n)\ze#+ '
```

![image](https://user-images.githubusercontent.com/79138/120499458-590b9300-c3c0-11eb-9566-31d48247bc8e.png)

This version would respect your md conceal for your code blocks. I assume you use `plasticboy/vim-markdown`
https://github.com/plasticboy/vim-markdown#fenced-code-block-languages

**Comment by laoshaw** [comment]
git checkout 9fb8ccc  does not work for me, checked all branches.

**Comment by sotte** [comment]
Glad it kinda works :tm: should be a smoother experience though! I'll try to fix this soon.

Out of curiosity: do you use the figlet feature?

**Comment by laoshaw** [comment]
I could not use figlet as I got "error: could not load font small" for all headings on each slide.

**Comment by laoshaw** [comment]
also when using 'figlet' I found for lists only the first level are figlet-ed, the second level of the list are still using whatever is from markdown, be it `*` or `-`

**Comment by laoshaw** [comment]
https://github.com/ikatyang/emoji-cheat-sheet/blob/master/README.md#symbols  maybe use emojis instead of figlet to make the slide more eye-candy? the title can just be bold

**Comment by sotte** [comment]
Related to #34

**Comment by gorrioncillo** [comment]
this same pass on python 
```python
#Comment
def hola():
  print('Hola mundo')

hola()
```
This make the slide divided into 2. The fixes here do not work.
I use vimwiki plugin too, so the markdown files seems to apperar as vimwiki type. do you think this the cause?

**Comment by jbargu** [comment]
I had the same issue and fixed it with:
1. `rm -rf vim-plug`.
2. Remove `tpope/vim-markdown` from the list of plugins.
3. Set `let g:vimwiki_global_ext = 0` for `vimwiki` to avoid `ft=vimwiki` on `*.md` files.

The crucial step seems to be removing the `vim-plug` dir.

**Comment by kavinvalli** [comment]
I am still facing this issue @sotte

**Comment by normen** [comment]
For me it works. I have no markdown plugin installed - tpopes plugin is integrated in VIM. I do have to add the highlight types to my `vimrc` though, like below. Those then also work in vimwiki though, which is a double-win.

```
let g:markdown_fenced_languages = ['go','perl','sh', 'python', 'cpp', 'c', 'javascript', 'vim']
```

**Comment by sotte** [comment]
My workaround is also not using any markdown plugin :) I should mention that in the README.

**Comment by rattletat** [comment]
For me, `vim-polyglot` caused my code cells to be not highlighted.

**Comment by VictorTaelin** [comment]
I managed to solve the issue here by commenting the lines that ident the code block to the right. Seems like doing so will cause my markdown highlighter to fail to detect the block as a code block. Can't work in a polished PR now, sorry, but here is a complete file:

https://gist.github.com/VictorTaelin/df15d860e1aca738794e6f4f67e17c48

Replace `autoload/markdown.vim` by it to solve the issue.
